src/dependency-manager/spec/component-spec.ts

```diff
@@ -39,8 +39,9 @@ export class IngressSpec {
 
   @IsOptional()
   @JSONSchema({
-    type: 'string',
+    ...ExpressionOr({ type: 'string', pattern: Slugs.ComponentSubdomainValidator.source }),
```

Comment:
Did you test deploying with a subdomain set via a secret?

This probably needs a test `interfaces.test.ts` to check the ingresses interpolation.

Reply:
added tests to interfaces to test both valid and invalid subdomain values passed through secrets

Reply:
Did you actually run `architect dev` to test?